### PR TITLE
Show judgement colours in hit distribution graph

### PR DESCRIPTION
**Pull request by acid-chicken**
Discussing in #9990

Changed the bars in the time distribution graph to stack for each `HitResult`.
Their colors are borrowed from those used in `HitErrorMeter`.

| osu!std | osu!mania | osu!taiko |
|-|-|-|
| std | mania | taiko |

### CODE REVIEW

**Comment by smoogipoo** [comment]
Without having gone into the code, design wise it looks pretty good to me.

I was thinking in order to handle hitobjects with different hit windows (see: https://github.com/ppy/osu/issues/9990#issuecomment-681943421), there could be a button at the bottom that lets you choose which hitobject to display. But it's not required for this implementation imo.

File: osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs [hunk 1]
```diff
@@ -207,52 +216,100 @@ private void updateDisplay()
             }
         }
 
+        private readonly struct BarValue
+        {
+            public readonly int Index;
```
Comment (peppy) [inline osu.Game/Screens/Ranking/Statistics/HitEventTimingDistributionGraph.cs:221]: This is pretty ugly

**Comment by peppy** [comment]
Overall this change adds quite a bit of what I would consider "too complex code" for what it's doing. Too many conversions between objects (ie. is `BarValue` needed? is the index flow proposed here worthwhile rather than just having a local function in `HitEventTimingDistributionGraph` or even `Bar` to handle it?).

But it seems to work okay and is mostly isolated, so maybe fine for now? Not sure.

**Comment by acid-chicken** [comment]
As you say, there is no point in having the index as a "value" and in going to the trouble of doing the process halfway through in each place, so I will put it all together in the Bar (after giving the structure the values needed for processing at the trial-and-error stage for the time being, I forgot to brush up on it).

File: osu.Game/Graphics/OsuColour.cs [hunk 2]
```diff
@@ -125,6 +125,34 @@ public Color4 ForHitResult(HitResult judgement)
             }
         }
 
+        public Color4 DrawForHitResult(HitResult result)
```
[the remark this answers is not in the document]
Comment (acid-chicken) [reply osu.Game/Graphics/OsuColour.cs:128]: I got it. FYI, these components will appear to be affected.

`BeatmapLeaderboard`
--

BeatmapLeaderboard

`ScorePanel`
--

ScorePanel

**Comment by peppy** [comment]
Thanks for making the changes. This is looking much better now.

**Review by peppy — APPROVED** [review]

**Pull request merged**